os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -87,28 +99,39 @@ void mm_initialize_ram_partitions(void)
 	 * The overhead value will include the size of mm_heap_s and the
 	 * allocation node and also a 100 byte buffer for future change in mm.
 	 */
-	uint32_t overhead = CONFIG_NUM_APPS * (SIZEOF_MM_ALLOCNODE + sizeof(struct mm_heap_s) + 100); 
+#ifdef CONFIG_MM_PARTITION_HEAP
+	uint32_t overhead = CONFIG_NUM_APPS * (SIZEOF_MM_ALLOCNODE + sizeof(struct mm_heap_s) + 100);
+#else
+	uint32_t overhead = 0;
+#endif
+
 #ifdef CONFIG_IMXRT_SEMC_SDRAM
-	uint32_t start = CONFIG_IMXRT_SDRAM_START + CONFIG_MM_KERNEL_HEAPSIZE;
-	uint32_t size = CONFIG_IMXRT_SDRAM_SIZE - CONFIG_MM_KERNEL_HEAPSIZE;
-	g_default_size = (size - overhead) / CONFIG_NUM_APPS;
+	g_ram_start = CONFIG_IMXRT_SDRAM_START + CONFIG_MM_KERNEL_HEAPSIZE;
+	g_ram_size = CONFIG_IMXRT_SDRAM_SIZE - CONFIG_MM_KERNEL_HEAPSIZE;
+#else
+	g_ram_start = (uint32_t)__usram_segment_start__ + (uint32_t)__usram_segment_size__;
+	g_ram_size = (uint32_t)(REGION_END - g_ram_start);
+#endif
 
-	DEBUGASSERT(start != 0);
-	DEBUGASSERT(size != 0);
+	g_default_size = (g_ram_size - overhead) / CONFIG_NUM_APPS;
+	DEBUGASSERT(g_ram_start != 0);
+	DEBUGASSERT(g_ram_size != 0);
 	DEBUGASSERT(g_default_size > 0);
 
-	mm_initialize(&g_pheap, (void *)start, size);
-	mvdbg("Default RAM partition size set to %u\n", g_default_size);
-	mvdbg("Initialized partition heap start = 0x%x size = %u\n", start, size);
+#ifdef CONFIG_MM_PARTITION_HEAP
+	mm_initialize(&g_pheap, (void *)g_ram_start, g_ram_size);
 #else
-	uint32_t start = (uint32_t)__usram_segment_start__ + (uint32_t)__usram_segment_size__;
+	int i;
+	for (i = 0; i < CONFIG_NUM_APPS; i++)
```

Comment:
Line 128 and line 129 are assumed that we have 2 apps.
But this for loop means we can have more than 2 apps. Is it okay?

Reply:
Yes. It does not look good. I will change it.